### PR TITLE
Improve consistency of documentation on context managers (code style)

### PR DESCRIPTION
**Pull request by Chillee**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #113285

Fixes https://github.com/pytorch/pytorch/pull/113285

### CODE REVIEW

**Review by albanD — COMMENTED** [review]
This is not a fix, both work just fine :D 
Do we really prefer users to do a call there?

**Comment by Chillee** [comment]
image

**Comment by albanD** [comment]
But the ones you're changing are decorators, not context managers :D

File: torch/autograd/grad_mode.py [hunk 1]
```diff
@@ -57,7 +57,7 @@ class no_grad(_NoParamDecoratorContextManager):
         >>> z = doubler(x)
         >>> z.requires_grad
         False
-        >>> @torch.no_grad
+        >>> @torch.no_grad()
```
Comment (jon-chuang) [inline torch/autograd/grad_mode.py:60]: @Chillee `@torch.no_grad` works fine. See: https://github.com/pytorch/pytorch/blob/d261687d5f56ac8148fab2567cf1fa6dd5264def/torch/utils/_contextlib.py#L120

I think this annotation is cleaner than with `()`

Comment (jon-chuang) [reply torch/autograd/grad_mode.py:60]: Hmm, that being said it seems that all uses of these in torch put the brackets, I guess changing makes sense to make the docs consistent with the usage.

**Comment by jon-chuang** [comment]
> This is not a fix, both work just fine :D
Do we really prefer users to do a call there?

I think this is more of a style fix - the rest of the docs and all of torch usage prefer to use `@torch.no_grad()` etc.

I don't think we have a preference - but for consistency we can just stick with calling.

**Review by jon-chuang — APPROVED** [review]
LGTM as it's a style fix that makes the docs more consistent

**Pull request closed without merging**